src/Perlin.tsx

```diff
@@ -30,10 +33,45 @@ class Perlin {
         return tx;
     }
 
+    public static getStoredHosts(): string[] {
+        const storedHosts = store.get(STORAGE_KEYS.STORED_HOSTS);
+        if (!storedHosts) {
+            Perlin.setStoredHosts([DEFAULT_HOST]);
```

Comment:
When our local storage does not have an entry for a custom stored host, I think we should just not save any data such as the default host.

Also, for storage-related functions we could put it in a separate class (to isolate Perlin as its own class with separate side effects from storage). Rest looks great to me.